code/weapon/weapons.cpp

```diff
@@ -7509,7 +7584,7 @@ void weapon_hit( object * weapon_obj, object * other_obj, vec3d * hitpos, int qu
 		}
 
 		particleSource.finish();
-	} else if (wip->dinky_impact_weapon_expl_effect.isValid() && !armed_weapon) {
+	} else if (!valid_conditional_impact && wip->dinky_impact_weapon_expl_effect.isValid() && !armed_weapon && !valid_conditional_impact) {
```

Comment:
Looks like `!valid_conditional_impact` is here twice

Reply:
Good catch! Thank you.